### PR TITLE
Fixes issue where block validation uses outdated chain metadata

### PR DESCRIPTION
**Pull request by neonknight64**
## Description
- These changes fixes an issue where the chain metadata is not correctly updated before the block validators make use of it, and would have resulted in invalid validation of blocks.
- Modified store_new_block to also update the backend metadata as well as the memory metadata when a block is added.

## Motivation and Context
Fixes a bug where the block validators are using outdated chain metadata, the difficulty adjustment algorithm would also not have had the correct information.

## How Has This Been Tested?
Existing tests remain functioning.

## Types of changes

* [x] Bug fix (non-breaking change which fixes an issue)
* [ ] New feature (non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [ ] Feature refactor (No new feature or functional changes, but performance or technical debt improvements)
* [ ] New Tests
* [ ] Documentation

## Checklist:

* [x] I'm merging against the `development` branch.
* [x] I ran `cargo-fmt --all` before pushing.
* [x] I have squashed my commits into a single commit.
* [ ] My change requires a change to the documentation.
* [ ] I have updated the documentation accordingly.
* [ ] I have added tests to cover my changes.

### CODE REVIEW

File: applications/tari_base_node/src/parser.rs [hunk 1]
```diff
@@ -675,7 +675,7 @@ impl Parser {
 
             let mut height = meta.height_of_longest_chain.expect("Could not retrieve chain height");
 
-            let mut current_header = node
+            let current_header = node
```
Comment (SWvheerden) [inline applications/tari_base_node/src/parser.rs:678]: you can remove this.
I think this stuck around after a refactor

File: base_layer/core/src/chain_storage/blockchain_database.rs [hunk 2]
```diff
@@ -730,16 +731,45 @@ fn add_block<T: BlockchainBackend>(
...
+        MetadataKey::ChainHeight,
+        MetadataValue::ChainHeight(Some(height)),
+    ));
+    txn.insert(DbKeyValuePair::Metadata(
+        MetadataKey::BestBlock,
+        MetadataValue::BestBlock(Some(best_block.clone())),
+    ));
+    txn.insert(DbKeyValuePair::Metadata(
+        MetadataKey::AccumulatedWork,
+        MetadataValue::AccumulatedWork(Some(accumulated_difficulty)),
```
Comment (SWvheerden) [inline base_layer/core/src/chain_storage/blockchain_database.rs:759]: Will this be reset after if reorg fails?

Comment (neonknight64) [reply base_layer/core/src/chain_storage/blockchain_database.rs:759]: Yes, then we will rewind the chain and add the original block back.

File: applications/tari_base_node/src/parser.rs (hunk 1 above)
Comment (neonknight64) [reply applications/tari_base_node/src/parser.rs:678]: Will remove in next PR.

**Pull request merged**